src/OpenSearch.Net.Auth.AwsSigV4/AwsSigV4HttpConnection.cs

```diff
@@ -0,0 +1,45 @@
+/* SPDX-License-Identifier: Apache-2.0
+*
+* The OpenSearch Contributors require contributions made to
+* this file be licensed under the Apache-2.0 license or a
+* compatible open source license.
+*/
+
+namespace OpenSearch.Net.Auth.AwsSigV4
+{
+	using System;
+	using Amazon;
+	using Amazon.Runtime;
+
+	public class AwsSigV4HttpConnection : HttpConnection
+	{
+		private readonly AWSCredentials _credentials;
+		private readonly RegionEndpoint _region;
+
+		public AwsSigV4HttpConnection(AWSCredentials credentials = null, RegionEndpoint region = null)
```

Comment:
Are credentials scoped to a region, and thus do we need those separately here?

Reply:
I'm not sure I understand your question or intent here? No, credentials aren't tied to a region, it'd be perfectly possible to do something like this contrived example:
```csharp
var credentials = new AssumeRoleAWSCredentials(
				FallbackCredentialsFactory.GetCredentials(),
				"arn:aws:iam::123456789012:role/my-open-search-ingest-role",
				"my-ingest-application");

var usClient = new OpenSearchClient(
                new ConnectionSettings(
                    new Uri("https://example-aaabbbcccddd111222333.us-east-1.es.amazonaws.com"), 
                    new AwsSigV4HttpConnection(credentials, RegionEndpoint.USEast1)));

var apClient = new OpenSearchClient(
                new ConnectionSettings(
                    new Uri("https://example-aaabbbcccddd111222333.ap-southeast-2.es.amazonaws.com"), 
                    new AwsSigV4HttpConnection(credentials, RegionEndpoint.APSoutheast2)));
```

Reply:
Can I just use `FallbackCredentialsFactory.GetCredentials()` as `credentials`? I *think* you need to get something with a region here, but I am not sure.

Reply:
Ah, with how the constructor was here it'd be `new AwsSigV4HttpConnection(region=RegionEndpoint.USEast1)`. But I've now made the additional overloads explicit for the constructor, so your example would work now.